### PR TITLE
Create

### PR DESCRIPTION
**Pull request by pjmiles**
Navbar created 
toggle function

### CODE REVIEW

File: src/components/sidebar/SideBar.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      path: "/",
+      name: "Blogs",
+      icon: <FaHome />,
+    },
+    {
+      path: "/create",
+      name: "CreateBlog",
+      icon: <ImBlogger2 />,
+    },
+  ];
```
Comment (Asylcreek) [inline src/components/sidebar/SideBar.js]: This can be outside of the Sidebar function

File: src/pages/BlogPage.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    if(window.innerHeight + e.target.documentElement.scrollTop + 1 >=
+      e.target.documentElement.scrollHeight) {
+        getBlog()
+      }
+  }
+
+  useEffect(() => {
+    getBlog();
+    window.addEventListener('scroll', handleScroll)
+  },[]);
```
Comment (Asylcreek) [inline src/pages/BlogPage.js]: `getBlog` definition should be moved inside this `useEffect`

There should be a cleanup function for the eventListener

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: src/components/sidebar/SideBar.js (hunk 1 above)
Comment (pjmiles) [reply src/components/sidebar/SideBar.js]: ok. I will adjust this.

File: src/pages/BlogPage.js (hunk 2 above)
Comment (pjmiles) [reply src/pages/BlogPage.js]: ok

File: src/components/sidebar/SideBar.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { useState } from "react";
+import { FaBars } from "react-icons/fa";
+import { NavLink } from "react-router-dom";
+import { menuItems } from "./MenuItems";
+import './SideBar.css'
+
+const SideBar = ({ children }) => {
+    const [isOpen, setIsOpen] = useState(false)
+    const toggle = () => setIsOpen(!isOpen)
```
Comment (Asylcreek) [inline src/components/sidebar/SideBar.js:9]: Use the updater function in `setIsOpen`

File: src/pages/BlogPage.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    if(window.innerHeight + e.target.documentElement.scrollTop + 1 >=
+      e.target.documentElement.scrollHeight) {
+        getBlog()
+      }
+  }
+
+  useEffect(() => {
+    getBlog();
+    window.addEventListener('scroll', handleScroll)
+  }, []);
```
Comment (Asylcreek) [inline src/pages/BlogPage.js]: This `useEffect` should have a cleanup function

File: src/pages/BlogPage.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  const loadMore = () => {
+    setPage(page + 1);
+  };
+
+  const handleScroll = (e) => {
+    if(window.innerHeight + e.target.documentElement.scrollTop + 1 >=
+      e.target.documentElement.scrollHeight) {
+        getBlog()
+      }
+  }
```
Comment (Asylcreek) [inline src/pages/BlogPage.js]: This should be moved into the `useEffect` below

File: src/pages/BlogPage.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  const [displayBlogs, setDisplayBlogs] = useState([]);
+  const [page, setPage] = useState(1);
+  let navigate = useNavigate()
+
+  const getBlog = async () => {
+    try {
+      const { data } = await axiosInstance.get();
+      const allPost = data.getBlogPost.docs
+      setDisplayBlogs((prevPost) => [...prevPost, ...allPost]);
+      loadMore()
```
Comment (Asylcreek) [inline src/pages/BlogPage.js]: After getting blogs, you load more blogs?

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: src/pages/BlogPage.js (hunk 4 above)
Comment (pjmiles) [reply src/pages/BlogPage.js]: ok this will looked into

File: src/pages/BlogPage.js (hunk 5 above)
Comment (pjmiles) [reply src/pages/BlogPage.js]: ok

File: src/pages/BlogPage.js (hunk 6 above)
Comment (pjmiles) [reply src/pages/BlogPage.js]: its just a function for the page, I can use the state function. I t will be adjusted

File: src/pages/BlogPage.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      if (
+        window.innerHeight + e.target.documentElement.scrollTop + 1 >=
+        e.target.documentElement.scrollHeight
+      ) {
+        getBlog();
+      }
+    };
+    getBlog();
+    window.addEventListener("scroll", handleScroll);
+  }, []);
```
Comment (Asylcreek) [inline src/pages/BlogPage.js]: This should have a `cleanup` function

File: src/pages/BlogPage.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  const getBlog = async () => {
+    try {
+      const { data } = await axiosInstance.get();
+      const allPost = data.getBlogPost.docs;
+      setDisplayBlogs((prevPost) => [...prevPost, ...allPost]);
+      setPage(page + 1);
+    } catch {
+      console.log("Error occured");
+    }
+  };
```
Comment (Asylcreek) [inline src/pages/BlogPage.js]: This can be moved into the `useEffect` below

File: src/pages/BlogPage.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  const [displayBlogs, setDisplayBlogs] = useState([]);
+  const [page, setPage] = useState(1);
+  let navigate = useNavigate();
+
+  const getBlog = async () => {
+    try {
+      const { data } = await axiosInstance.get();
+      const allPost = data.getBlogPost.docs;
+      setDisplayBlogs((prevPost) => [...prevPost, ...allPost]);
+      setPage(page + 1);
```
Comment (Asylcreek) [inline src/pages/BlogPage.js]: Why?

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: src/pages/BlogPage.js (hunk 7 above)
Comment (pjmiles) [reply src/pages/BlogPage.js]: now added by removing the event on window

File: src/pages/BlogPage.js (hunk 8 above)
Comment (pjmiles) [reply src/pages/BlogPage.js]: adjusted

File: src/pages/BlogPage.js [hunk 10]
```diff
@@ -8,18 +8,19 @@ const BlogPage = () => {
...
-  };
-
+ 
   useEffect(() => {
+    const getBlog = async () => {
+      try {
+        const { data } = await axiosInstance.get();
+        const allPost = data.getBlogPost.docs;
+        setDisplayBlogs((prevPost) => [...prevPost, ...allPost]);
+        setPage();
```
Comment (Asylcreek) [inline src/pages/BlogPage.js]: Why?

**Review by Asylcreek — CHANGES_REQUESTED** [review]

File: src/pages/BlogPage.js (hunk 9 above)
Comment (pjmiles) [reply src/pages/BlogPage.js]: this is to render the blogs from the backend

Comment (Asylcreek) [reply src/pages/BlogPage.js]: Why call `setPage` after fetching?

And where do you use the `page` state variable?

Comment (pjmiles) [reply src/pages/BlogPage.js]: > Why call `setPage` after fetching?
> 
> And where do you use the `page` state variable?
 `page` state variable is used as an argument inside the useEffect dep. array.

Comment (pjmiles) [reply src/pages/BlogPage.js]: I have removed the page state totally, allowed the widows to handle the scroll.

File: src/pages/BlogPage.js (hunk 10 above)
Comment (pjmiles) [reply src/pages/BlogPage.js]: adjusted

**Review by Asylcreek — APPROVED** [review]

**Pull request merged**